packages/sitecore-jss-dev-tools/src/package-deploy.ts

```diff
@@ -182,99 +163,133 @@ async function watchJobStatus(options: PackageDeployOptions, taskName: string) {
   const factors = [options.appName, taskName, `${options.importServiceUrl}/status`];
   const mac = hmac(factors, options.secret);
 
-  const requestBaseOptions = {
+  const url = new URL(
+    `${options.importServiceUrl}/status?appName=${options.appName}&jobName=${taskName}&after=${logOffset}`
+  );
+
+  const isHttps = options.importServiceUrl.startsWith('https');
+  const client = isHttps ? https : http;
+
+  const reqOptions: RequestOptions = {
     headers: {
       'User-Agent': 'Sitecore/JSS-Import',
       'Cache-Control': 'no-cache',
       'X-JSS-Auth': mac,
     },
-    dispatcher: new ProxyAgent({
-      uri: options.proxy || '',
-      maxRedirections: 0,
-      connect: {
-        rejectUnauthorized: options.acceptCertificate ? false : true,
-        maxCachedSessions: options.acceptCertificate ? 0 : undefined,
-      },
-    }),
+    method: 'GET',
+    hostname: url.hostname,
+    protocol: url.protocol,
+    path: url.pathname + url.search,
+    agent: isHttps
+      ? new https.Agent({
+          // we turn off normal CA cert validation when we are whitelisting a single cert thumbprint
+          rejectUnauthorized: options.acceptCertificate ? false : true,
+          // needed to allow whitelisting a cert thumbprint if a connection is reused
+          maxCachedSessions: options.acceptCertificate ? 0 : undefined,
+        })
+      : undefined,
   };
 
   if (options.debugSecurity) {
     console.log(`Deployment status security factors: ${factors}`);
     console.log(`Deployment status HMAC: ${mac}`);
   }
 
-  return new Promise((resolve, reject) => {
-    /**
-     * Send job status request
-     */
-    function sendJobStatusRequest() {
-      new NativeDataFetcher()
-        .get<JobStatusResponse>(
-          `${options.importServiceUrl}/status?appName=${options.appName}&jobName=${taskName}&after=${logOffset}`,
-          requestBaseOptions
+  if (options.proxy) {
+    setProxy(reqOptions, options.proxy, options.importServiceUrl);
+  }
+
+  const req = client.request(reqOptions);
+
+  let responseData = '';
+
+  req.on('response', (res) => {
```

Comment:
it's a pain, but we need to unit test these handlers and ensure data is being read ok, and errors are being handled okay too

Reply:
I added a bit of unit tests to cover `formData` events, let's add more unit tests once we come back to increase coverage, since `packageDeploy` functionality wasn't covered at all